sessionctx/variable/sysvar.go

```diff
@@ -693,6 +695,16 @@ var defaultSysVars = []*SysVar{
 	}, GetGlobal: func(s *SessionVars) (string, error) {
 		return fmt.Sprint(QueryLogMaxLen.Load()), nil
 	}},
+	{Scope: ScopeGlobal, Name: TiDBCommitterConcurrency, Value: strconv.Itoa(DefTiDBCommitterConcurrency), Type: TypeInt, MinValue: 1, MaxValue: 10000, SetGlobal: func(s *SessionVars, val string) error {
+		tikvConfigLock.Lock()
+		defer tikvConfigLock.Unlock()
```

Comment:
Lock this code block only takes no effect, the other [function](https://github.com/pingcap/tidb/blob/f88dab92f9c5ca3f2225a1dab19ff9b4e6aaaaa1/config/config.go#L901) stores config to TiKV client should also be locked.

Reply:
@espresso98 maybe we can move the mutex from variable package to config package so the one lock can protect both. i.e. 
```
$ git diff
diff --git a/config/config.go b/config/config.go
index 4cc572b0a..c8d373269 100644
--- a/config/config.go
+++ b/config/config.go
@@ -24,6 +24,7 @@ import (
        "os/user"
        "path/filepath"
        "strings"
+       "sync"
        "sync/atomic"
        "time"
 
@@ -141,6 +142,9 @@ var (
        // DeprecatedOptions indicates the config options existing in some other sections in config file.
        // They should be moved to [instance] section.
        DeprecatedOptions []InstanceConfigSection
+
+       // TikvConfigLock protects against concurrent tikv config refresh
+       TikvConfigLock sync.Mutex
 )
 
 // Config contains configuration options.
@@ -902,6 +906,8 @@ func GetGlobalConfig() *Config {
 // StoreGlobalConfig stores a new config to the globalConf. It mostly uses in the test to avoid some data races.
 func StoreGlobalConfig(config *Config) {
        globalConf.Store(config)
+       TikvConfigLock.Lock()
+       defer TikvConfigLock.Unlock()
        cfg := *config.GetTiKVConfig()
        tikvcfg.StoreGlobalConfig(&cfg)
 }
diff --git a/sessionctx/variable/sysvar.go b/sessionctx/variable/sysvar.go
index e74f53fa8..81225b47a 100644
--- a/sessionctx/variable/sysvar.go
+++ b/sessionctx/variable/sysvar.go
@@ -696,8 +696,8 @@ var defaultSysVars = []*SysVar{
                return fmt.Sprint(QueryLogMaxLen.Load()), nil
        }},
        {Scope: ScopeGlobal, Name: TiDBCommitterConcurrency, Value: strconv.Itoa(DefTiDBCommitterConcurrency), Type: TypeInt, MinValue: 1, MaxValue: 10000, SetGlobal: func(s *SessionVars, val string) error {
-               tikvConfigLock.Lock()
-               defer tikvConfigLock.Unlock()
+               config.TikvConfigLock.Lock()
+               defer config.TikvConfigLock.Unlock()
                tikvutil.CommitterConcurrency.Store(int32(TidbOptInt64(val, DefTiDBCommitterConcurrency)))
                cfg := config.GetGlobalConfig().GetTiKVConfig()
                tikvcfg.StoreGlobalConfig(cfg)
diff --git a/sessionctx/variable/tidb_vars.go b/sessionctx/variable/tidb_vars.go
index f9a8c2acf..7aefb6296 100644
--- a/sessionctx/variable/tidb_vars.go
+++ b/sessionctx/variable/tidb_vars.go
@@ -16,7 +16,6 @@ package variable
 
 import (
        "math"
-       "sync"
 
        "github.com/pingcap/tidb/config"
        "github.com/pingcap/tidb/parser/mysql"
@@ -894,8 +893,7 @@ var (
        StatsLoadPseudoTimeout                = atomic.NewBool(DefTiDBStatsLoadPseudoTimeout)
        MemQuotaBindingCache                  = atomic.NewInt64(DefTiDBMemQuotaBindingCache)
        GCMaxWaitTime                         = atomic.NewInt64(DefTiDBGCMaxWaitTime)
-       tikvConfigLock                 sync.Mutex
-       StatsCacheMemQuota             = atomic.NewInt64(DefTiDBStatsCacheMemQuota)
+       StatsCacheMemQuota                    = atomic.NewInt64(DefTiDBStatsCacheMemQuota)
 )
 
 var (

```

Reply:
Thanks, Morgan!!